Build failure: python311Packages.aplpy

### Steps To Reproduce

Steps to reproduce the behavior:
1. build python311Packages.plpy

### Build log

Several tests are failing with one of those two errors:
```
...
FAILED aplpy/tests/test_misc.py::test_stretches - AttributeError: module 'matplotlib.cm' has no attribute 'get_cmap'
...
FAILED aplpy/tests/test_pixworldmarkers.py::test_wcs_coords[inputval0] - AttributeError: 'TimeAttribute' object has no attribute 
'jyear'
...
```

### Notify maintainers
cc @smaret 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.37, NixOS, 24.11 (Vicuna), 24.11.20240716.ad0b5ee`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - nixpkgs: `/nix/store/70y8v9qfn34sdfsnwh3l9772lk4rpfiy-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/aplpy/default.nix
+++ b/pkgs/development/python-modules/aplpy/default.nix
@@ -33,7 +33,7 @@ buildPythonPackage rec {
     hash = "sha256-KCdmBwQWt7IfHsjq7pWlbSISEpfQZDyt+SQSTDaUCV4=";
   };
 
-  # Fix build with Astropy 6 and Python 3.12.
+  # Fix build with Astropy 6.1, Python 3.12 and matplotlib 3.9
   patches = [
     # https://github.com/aplpy/aplpy/pull/496
     (fetchpatch {
@@ -44,6 +44,16 @@ buildPythonPackage rec {
     (fetchpatch {
       url = "https://github.com/aplpy/aplpy/commit/468be394970b39f1aaa6debef51eb674e2dd86d8.patch";
       hash = "sha256-/ovLrFOKb3RQ8TZSviuOV6EYNgz0gtrhVWZLFJBrzFg=";
+    })
+    # https://github.com/aplpy/aplpy/pull/500
+    (fetchpatch {
+      url = "https://github.com/aplpy/aplpy/commit/c9049261b3724821f4d309d85fe73cabcc04d890.patch";
+      hash = "sha256-ivdXLh1Z+WG3EL5u1wO+lqz6X1AP6WpyoZix5YiqW0w=";
+    })
+    # https://github.com/aplpy/aplpy/pull/503
+    (fetchpatch {
+      url = "https://github.com/aplpy/aplpy/commit/edc2aa9770b72bd3ac42db95a6e027b276be9c31.patch";
+      hash = "sha256-QUZenY3XQ/T2tl5apxZUqjf4ACjOzJWb+1AUp5limi4=";
     })
   ];
 
@@ -81,9 +91,5 @@ buildPythonPackage rec {
     homepage = "http://aplpy.github.io";
     license = licenses.mit;
     maintainers = with maintainers; [ smaret ];
-    # Tests fail due to expired matplotlib deprecations:
-    #
-    # > (`matplotlib.cm' has no attribute 'get_cmap'`)
-    broken = true;
   };
 }

